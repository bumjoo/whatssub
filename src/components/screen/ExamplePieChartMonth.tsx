import React, { Component } from 'react';
import {
  Platform,
  StatusBar,
  StyleSheet,
  TouchableHighlight,
  Image,
  ScrollView,
  Text,
  View,
  FlatList,
  InteractionManager,
  TouchableOpacity,
} from 'react-native';

import styled from 'styled-components/native';

import PieChart, { IProduct } from '../shared/PieChart';

import Button from '../shared/Button';
import { getString } from '../../../STRINGS';

const Container = styled.View`
  flex: 1;
  background-color: ${(props) => props.theme.backgroundDark};
  align-items: center;
  justify-content: flex-start;
`;

interface IProps {
  navigation: any;
}

function Page(props: IProps) {
  const [data, setData] = React.useState(null);
  const [selectedMonth, setSelectedMonth] = React.useState('2019.06');
  React.useEffect(() => {
    setData(sampleDatas[selectedMonth]);
    return () => {};
  }, [selectedMonth]);

  return (
    <Container style={{ alignItems: 'center', justifyContent: 'center', backgroundColor: 'white' }}>
      <PieChart
        data={data}
        style={{ width: 300, height: 300 }}
        defaultShowSlice={ 0 }
        currentMonth={selectedMonth}
      />
      {/* <PieChartWithDynamicSlices/>
      <PieChartWithCenteredLabels/> */}
      <TouchableOpacity
        onPress={ () => {
          setSelectedMonth('2019.06');
        }}
      >
        <Text>{'2019.06'}</Text>
      </TouchableOpacity>
      <TouchableOpacity
        onPress={ () => {
          setSelectedMonth('2019.07');
        }}
      >
        <Text>{'2019.07'}</Text>
      </TouchableOpacity>
    </Container>
  );
}

export default Page;

const sampleDatas: { [ key : string ]: IProduct[] } = {
  2019.06: [
    {
      name: 'prod_1',
      category: getString('CATEGORY_CULTURE_LEISURE'),
      price: 100,
      currency: '',
      date: new Date(),
      image: '',
      currentMonthPaymentDate: new Date(),
    },
    {
      name: 'prod_2',
      category: getString('CATEGORY_CULTURE_LEISURE'),
      price: 100,
      currency: '',
      date: new Date(),
      image: '',
      currentMonthPaymentDate: new Date(),
    },
    {
      name: 'prod_3',
      category: getString('CATEGORY_CULTURE_LEISURE'),
      price: 100,
      currency: '',
      date: new Date(),
      image: '',
      currentMonthPaymentDate: new Date(),
    },
    {
      name: 'prod_4',
      category: getString('CATEGORY_EDUCATION'),
      price: 100,
      currency: '',
      date: new Date(),
      image: '',
      currentMonthPaymentDate: new Date(),
    },
    {
      name: 'prod_5',
      category: getString('CATEGORY_GAME'),
      price: 100,
      currency: '',
      date: new Date(),
      image: '',
      currentMonthPaymentDate: new Date(),
    },
  ],
  2019.07: [
    {
      name: 'prod_1',
      category: getString('CATEGORY_CULTURE_LEISURE'),
      price: 100,
      currency: '',
      date: new Date(),
      image: '',
      currentMonthPaymentDate: new Date(),
    },
    {
      name: 'prod_3',
      category: getString('CATEGORY_CULTURE_LEISURE'),
      price: 200,
      currency: '',
      date: new Date(),
      image: '',
      currentMonthPaymentDate: new Date(),
    },
    {
      name: 'prod_4',
      category: getString('CATEGORY_EDUCATION'),
      price: 100,
      currency: '',
      date: new Date(),
      image: '',
      currentMonthPaymentDate: new Date(),
    },
    {
      name: 'prod_5',
      category: getString('CATEGORY_GAME'),
      price: 100,
      currency: '',
      date: new Date(),
      image: '',
      currentMonthPaymentDate: new Date(),
    },
    {
      name: 'prod_6',
      category: getString('CATEGORY_GAME'),
      price: 200,
      currency: '',
      date: new Date(),
      image: '',
      currentMonthPaymentDate: new Date(),
    },

  ],
};
