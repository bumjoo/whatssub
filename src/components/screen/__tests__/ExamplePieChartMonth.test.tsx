import 'react-native';
import * as React from 'react';
import ExamplePieChartMonth from '../ExamplePieChartMonth';

import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components/native';
import { render, fireEvent, act, RenderResult } from '@testing-library/react-native';

import { createTheme, ThemeType } from '../../../theme';

const createTestProps = (obj: object) => ({
  navigation: {
    navigate: jest.fn(),
  },
  ...obj,
});
const props: any = createTestProps({});
const component = (
  <ThemeProvider theme={createTheme(ThemeType.LIGHT)}>
    <ExamplePieChartMonth {...props} />
  </ThemeProvider>
);

describe('[ExamplePieChartMonth]', () => {
  it('renders without crashing', () => {
    const rendered = renderer.create(component).toJSON();
    expect(rendered).toMatchSnapshot();
    expect(rendered).toBeTruthy();
  });
});
