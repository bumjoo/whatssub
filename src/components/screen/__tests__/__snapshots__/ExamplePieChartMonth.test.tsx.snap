// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[ExamplePieChartMonth] renders without crashing 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#023059",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "flex-start",
      },
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "justifyContent": "center",
      },
    ]
  }
>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Text>
      2019.06
    </Text>
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Text>
      2019.07
    </Text>
  </View>
</View>
`;
